src/sidebar/components/tag-editor.js

```diff
@@ -87,96 +95,90 @@ function TagEditor({ onEditTags, tags: tagsService, tagList }) {
       return;
     }
     updateTags([...tagList, value]);
-    setShowSuggestions(false);
+    setOpen(false);
 
     inputEl.current.value = '';
     inputEl.current.focus();
   };
 
   /**
-   *  If the user pressed enter or comma while focused on
-   *  the <input>, then add a new tag.
+   *  Update the suggestions if the user changes the value of the input
    */
-  const handleKeyPress = e => {
-    if (e.key === 'Enter' || e.key === ',') {
-      addTag();
-      // preventDefault stops the delimiter from being
-      // added to the input field
-      e.preventDefault();
-    }
-  };
-
   const handleOnInput = e => {
     if (
       e.inputType === 'insertText' ||
       e.inputType === 'deleteContentBackward'
     ) {
-      if (inputEl.current.value.length === 0) {
-        // If the user deleted input, hide suggestions. This has
-        // no effect in Safari and the list will stay open.
-        setShowSuggestions(false);
-      } else {
-        if (isChromeUA) {
-          // Filter down the suggestions list for Chrome to subvert a performance
-          // issue. Returning too many suggestions causes a noticeable delay when
-          // rendering the datalist.
-          // See https://github.com/hypothesis/client/issues/1606
-          setSuggestionsFiltered(
-            removeDuplicates(
-              tagsService.filter(inputEl.current.value, chromeSuggestionsLimit),
-              tagList
-            )
-          );
-        }
-        // Show the suggestions if the user types something into the field
-        setShowSuggestions(true);
-      }
-    } else if (
-      e.inputType === undefined ||
-      e.inputType === 'insertReplacementText'
-    ) {
-      // nb. Chrome / Safari reports undefined and Firefox reports 'insertReplacementText'
-      // for the inputTyp value when clicking on an element in the datalist.
-      //
-      // There are two ways to arrive here, either click an item with the mouse
-      // or use keyboard navigation and press 'Enter'.
-      //
-      // If the input value typed already exactly matches the option selected
-      // then this event won't fire and a user would have to press 'Enter' a second
-      // time to trigger the handleKeyPress callback above to add the tag.
-      // Bug: https://github.com/hypothesis/client/issues/1604
-      addTag();
+      updateSuggestions();
+    }
+  };
+
+  /**
+   *  Callback when the user clicked one of the items in the suggestions list.
+   *  This will add a new tag.
+   */
+  const handleSelect = item => {
+    if (item) {
```

Comment:
we could get away w/o this condition, but addTag does not check for empty objects, I could go either way on this